src/components/LanguageButton/LanguageButton.tsx

```diff
@@ -1,19 +1,19 @@
 import { useContext, FunctionComponent as FC } from 'react';
 import cn from 'classnames';
 import { PageLanguageContext } from 'src/contexts';
-import { createLanguageHrefFromDefaults, getLanguageDefaults, getFilteredLanguages } from 'src/components';
+import { createLanguageHrefFromDefaults, getLanguageDefaults, getTrimmedLanguage } from 'src/components';
 import languageLabels from 'src/maps/language';
 import { cacheVisitPreferredLanguage } from 'src/utilities';
 import { LanguageNavigationComponentProps } from '../Menu/LanguageNavigation';
 import { button, isActive } from '../Menu/MenuItemButton/MenuItemButton.module.css';
 
-const LanguageButton: FC<LanguageNavigationComponentProps> = ({ language }) => {
+const LanguageButton: FC<LanguageNavigationComponentProps> = ({ language, selectedLanguageForPre }) => {
   const pageLanguage = useContext(PageLanguageContext);
-  const selectedLanguage = getFilteredLanguages(language);
-  const { isLanguageDefault, isPageLanguageDefault, isLanguageActive } = getLanguageDefaults(
-    selectedLanguage,
-    pageLanguage,
-  );
+  const selectedLanguage = getTrimmedLanguage(language);
+  const { isLanguageDefault, isPageLanguageDefault } = getLanguageDefaults(selectedLanguage, pageLanguage);
+  /* separated the isLanguageActive as we will pass a pageLanguage that is not only from the context lang, but a condition
```

Comment:
Sorry, I don't understand this comment - what is the context lang?

Reply:
@MymmiJ   I updated the comment, hopefully now its easy to understand